common/chunkStatusLogger.go

```diff
@@ -324,17 +321,6 @@ func (csl *chunkStatusLogger) CloseLogger() {
 	// Once logger is closed, we log no more chunks.
 	csl.outputEnabled = false
 
-	// lets make sure no previous entry exist in channel.
-	for {
-		if len(csl.unsavedEntries) == 0 {
-			break
-		}
-		time.Sleep(10 * time.Millisecond)
-	}
-
-	// Signal main logger goroutine to packup.
-	csl.done <- struct{}{}
-
 	// No more chunks will ever be written, let the main logger know about this.
```

Comment:
add a comment here

// On closing this channel the main logger will exit from its for-range loop.